ovs/lib/albacontroller.py

```diff
@@ -1116,13 +1116,29 @@ def _remove_services(ip, alba_backend):
         Stops and removes the maintenance service/process
         """
         client = SSHClient(ip, username='root')
-        for service_prefix in [AlbaController.ALBA_MAINTENANCE_SERVICE_PREFIX, AlbaController.ALBA_REBALANCER_SERVICE_PREFIX]:
-            service_name = '{0}_{1}'.format(service_prefix, alba_backend.backend.name)
+        service_name = '{0}_{1}'.format(AlbaController.ALBA_MAINTENANCE_SERVICE_PREFIX, alba_backend.backend.name)
+        if ServiceManager.has_service(service_name, client=client) is True:
+            if ServiceManager.get_service_status(service_name, client=client) is True:
+                ServiceManager.stop_service(service_name, client=client)
+            ServiceManager.remove_service(service_name, client=client)
+
+    @staticmethod
+    @add_hooks('update', 'postupgrade')
+    def upgrade_alba_plugin(client):
+
+        from ovs.dal.lists.albabackendlist import AlbaBackendList
+        alba_backends = AlbaBackendList.get_albabackends()
+        for alba_backend in alba_backends:
+            alba_backend_name = alba_backend.backend.name
+            service_name = '{0}_{1}'.format(AlbaController.ALBA_REBALANCER_SERVICE_PREFIX, alba_backend_name)
             if ServiceManager.has_service(service_name, client=client) is True:
                 if ServiceManager.get_service_status(service_name, client=client) is True:
                     ServiceManager.stop_service(service_name, client=client)
                 ServiceManager.remove_service(service_name, client=client)
 
+            AlbaController._setup_services(client.ip, alba_backend_name + '-abm', alba_backend_name)
```

Comment:
setup services requires the abm name and alba_backend itself, so this wont work since you pass a string twice
2nd: i'd rewrite the _setup_services to only require the storagerouter IP and the albabackend object
With that albabackend object passed, the name can be retrieved